app/src/main/java/org/mozilla/fenix/settings/SyncPreference.kt

```diff
@@ -0,0 +1,41 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.View
+import androidx.appcompat.widget.SwitchCompat
+import androidx.preference.PreferenceViewHolder
+import androidx.preference.SwitchPreferenceCompat
+import org.mozilla.fenix.R
+
+/**
+ * Variation of [SwitchPreferenceCompat] that uses a custom widgetLayoutResource in order to implement
+ * visibility changes to it.
+ * */
+class SyncPreference @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null
+) : SwitchPreferenceCompat(context, attrs) {
+
+    private var switchView: SwitchCompat? = null
+    var widgetVisible: Boolean = false
```

Comment:
Maybe we should call this `isSwitchWidgetVisible`, `isSwitchVisible` or some variation of this that confirms to Android naming convention.